sql/binlog.h

```diff
@@ -883,6 +884,23 @@ class MYSQL_BIN_LOG : public TC_LOG {
     mysql_mutex_assert_not_owner(&LOCK_log);
     return atomic_binlog_end_pos;
   }
+
+  my_off_t get_binlog_end_pos_async(bool async_slave) const {
+    mysql_mutex_assert_not_owner(&LOCK_log);
+    if (async_slave) return last_acked.second;
+    return atomic_binlog_end_pos;
+  }
+
+  void signal_semi_sync_ack(const char *const log_file,
```

Comment:
Is there a way to push this inside the plugin as well?

Reply:
As above